src/utils/github/getAuthors.ts

```diff
@@ -26,6 +28,137 @@ export async function getAuthors(
     });
   } catch (err) {
     // eslint-disable-next-line no-console
+    // @ts-ignore
+    console.error(err);
+    return [];
+  }
+}
+
+interface RevertAuthorInfo {
+  login: string | null;
+  email: string | null;
+}
+
+function extractRevertedCommitHash(commitMessage: string): string | null {
+  const match = commitMessage.match(/This reverts commit ([a-f0-9]+)/i);
+  return match ? match[1] : null;
+}
+
+async function getRevertCommitDetails(
+  revertedHash: string,
+  owner: string,
+  repo: string,
+  octokitInstance: Octokit
+): Promise<RevertAuthorInfo | null> {
+  try {
+    const revertedCommitDetailsResponse = await octokitInstance.request(
+      'GET /repos/{owner}/{repo}/commits/{ref}',
+      {
+        owner,
+        repo,
+        ref: revertedHash,
+        headers: { 'X-GitHub-Api-Version': '2022-11-28' },
+      }
+    );
+    const revertedData = revertedCommitDetailsResponse.data;
+
+    const revertedCommitAuthor = revertedData.author
+      ? {
+          login: revertedData.author.login,
+          email: revertedData.commit.author?.email ?? null,
+        }
+      : null;
+
+    return revertedCommitAuthor;
+  } catch (revertError) {
+    console.error(
+      `  Error fetching details for reverted commit ${revertedHash}:`,
+      revertError
+    );
+    return null;
+  }
+}
+
+async function processCommit(
+  commitSha: string,
+  owner: string,
+  repo: string,
+  octokitInstance: Octokit
+): Promise<RevertAuthorInfo | null> {
+  try {
+    const commitDetailsResponse = await octokitInstance.request(
+      'GET /repos/{owner}/{repo}/commits/{ref}',
+      {
+        owner,
+        repo,
+        ref: commitSha,
+        headers: { 'X-GitHub-Api-Version': '2022-11-28' },
+      }
+    );
+    const commitData = commitDetailsResponse.data;
+    const revertedHash = extractRevertedCommitHash(commitData.commit.message);
+    let revertDetails: RevertAuthorInfo | null = null;
+
+    if (revertedHash) {
+      revertDetails = await getRevertCommitDetails(
+        revertedHash,
+        owner,
+        repo,
+        octokitInstance
+      );
+    }
+
+    return revertDetails;
+  } catch (error) {
+    console.error(`Error processing commit ${commitSha}:`, error);
+    return null;
+  }
+}
+
+export async function getAuthorsWithRevertedCommitAuthors(
```

Comment:
This isn't ever used anywhere. Also, instead of having this be a separate function, we should make it a flag passed into the getAuthors function.